src/common/grpc.rs

```diff
@@ -1042,6 +1156,27 @@ where
         self.encode_message(resp)
     }
 
+    fn power_sensor_do_command(&mut self, message: &[u8]) -> Result<(), ServerError> {
```

Comment:
It feels a little wrong to me to need 7 different implementations of this. Is there a way on robot that we could support uniformly getting the access to the `DoCommand` interface of these different types? As far as I can see the only actual difference between, say, `power_sensor_do_command` and `encoder_do_command` is the method called on `.robot`.